File: C++/2763. Sum of Imbalance Numbers of All Subarrays.cpp
class Solution {
public:
    int sumImbalanceNumbers(vector<int>& nums) {
        int n = nums.size(), res = 0;

        for (int i = 0; i < n; i++) {
            set<int> s;
            int t = 0;
            s.insert(nums[i]);

            for (int j = i + 1; j < n; j++) {
                if (s.count(nums[j]) != 0);
                else if (s.count(nums[j] + 1) == 0 && s.count(nums[j] - 1) == 0) t += 1;
                else if (s.count(nums[j] + 1) != 0 && s.count(nums[j] - 1) != 0) t -= 1;

                s.insert(nums[j]);
                res += t;
            }
        }

        return res;
    }
};

File: Python/2763. Sum of Imbalance Numbers of All Subarrays.py
class Solution:
    def sumImbalanceNumbers(self, nums: List[int]) -> int:
        n, res = len(nums), 0

        for i in range(n):
            s, t = set(), 0
            s.add(nums[i])

            for j in range(i + 1, n):
                if nums[j] in s: pass
                elif nums[j] + 1 not in s and nums[j] - 1 not in s: t += 1
                elif nums[j] + 1 in s and nums[j] - 1 in s: t -= 1

                s.add(nums[j])
                res += t

        return res
